### ISSUE TITLE
`disableRequestLogging` should disable logs in default error handler

### ISSUE THREAD

**Issue by SamSalvatico** [fastify/fastify#5409]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.26.2

### Node.js version

v20.11.1

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

14.4.1

### Description

I am trying to have a full control over the log entries Fastify writes and I noticed that in the default error handler it [writes records](https://github.com/fastify/fastify/blob/main/lib/error-handler.js#L82) without checking the `disableRequestLogging` value.

Should it check if and disable logging when it is `true` or is it preferred to set a custom error handler to prevent it?

### Steps to Reproduce

```
import fastify from 'fastify';

const server = fastify({
    disableRequestLogging: true
});

server.get('/ping', async (request, reply) => {
    throw new Error('err');
})

server.listen({ port: 8080 }, (err, address) => {
    if (err) {
        process.exit(1)
    }
})
```

### Expected Behavior

No log entries added when an error is managed by the default error handler

**Comment by metcoder95** [comment]
Setting a custom `errorHandler` should provide you the option to disable the logging.

Though, I do see the benefit when you might not want to customize the `errorHandler` but just disable the logging. Might be a good enhancement

**Comment by SamSalvatico** [comment]
> Setting a custom `errorHandler` should provide you the option to disable the logging.
> 
> Though, I do see the benefit when you might not want to customize the `errorHandler` but just disable the logging. Might be a good enhancement

Could I then create a PR related to this?
Thank you for you time

**Comment by metcoder95** [comment]
Please, PR always welcomed

**Comment by SamSalvatico** [comment]
> Please, PR always welcomed

Great, I'm working on it

Would you like also to apply the same behaviour to the [404 handler](https://github.com/fastify/fastify/blob/main/lib/fourOhFour.js#L53)?

**Comment by metcoder95** [comment]
Sure, I believe the configuration should apply widely 👍

**Comment by metcoder95** [comment]
Fixed through #5435
